assets/js/modules/adsense/util/url.js

```diff
@@ -0,0 +1,141 @@
+ */
+export const getAccountURL = ( { accountID, userEmail } = {} ) => {
+	const baseURL = getAccountBaseURL( { accountID, userEmail, path: '/home' } );
+	return addQueryArgs( baseURL, { source: 'site-kit' } );
+};
+
+/**
+ * Returns the URL to an AdSense account's site overview page.
+ *
+ * @since n.e.x.t
+ *
+ * @param {Object}             options           Options for generating the URL.
+ * @param {(string|undefined)} options.accountID The AdSense account ID.
+ * @param {(string|undefined)} options.siteURL   The site URL to link to in AdSense.
+ * @param {(string|undefined)} options.userEmail The Google account email address. Relevant
+ *                                    for users with multiple Google accounts.
+ * @return {string} AdSense account site overview URL.
+ */
+export const getAccountSiteURL = ( { accountID, siteURL, userEmail } = {} ) => {
+	const baseURL = getAccountBaseURL( { accountID, userEmail, path: '/sites/my-sites' } );
+	const queryParams = { source: 'site-kit' };
+	if ( siteURL ) {
+		queryParams.url = parseDomain( siteURL ) || siteURL;
+	}
+	return addQueryArgs( baseURL, queryParams );
+};
+
+/**
+ * Returns the URL to an AdSense account's site ads preview page.
+ *
+ * @since n.e.x.t
+ *
+ * @param {Object}             options           Options for generating the URL.
+ * @param {(string|undefined)} options.accountID The AdSense account ID.
+ * @param {(string|undefined)} options.siteURL   The site URL to link to in AdSense.
+ * @param {(string|undefined)} options.userEmail The Google account email address. Relevant
+ *                                    for users with multiple Google accounts.
+ * @return {string} AdSense account site ads preview URL.
+ */
+export const getAccountSiteAdsPreviewURL = ( { accountID, siteURL, userEmail } = {} ) => {
+	const baseURL = getAccountBaseURL( { accountID, userEmail, path: '/myads/sites/preview' } );
+	const queryParams = { source: 'site-kit' };
+	if ( siteURL ) {
+		queryParams.url = parseDomain( siteURL ) || siteURL;
+	}
+	return addQueryArgs( baseURL, queryParams );
+};
+
+/**
+ * Returns a URL to an account screen in the AdSense frontend.
+ *
+ * @since n.e.x.t
+ *
+ * @param {Object}             options           Options for generating the URL.
+ * @param {(string|undefined)} options.accountID The AdSense account ID.
+ * @param {(string|undefined)} options.userEmail The Google account email address. Relevant
+ *                                    for users with multiple Google accounts.
+ * @param {(string|undefined)} options.path      Additional path after the base URL generated.
+ *                                    Only relevant if accountID is also provided.
+ * @return {string} AdSense frontend URL.
+ */
+const getAccountBaseURL = ( { accountID, userEmail, path = '/home' } = {} ) => {
+	let baseURL = 'https://www.google.com/adsense/new';
+	if ( userEmail ) { // Allows to indicate which Google account to use.
+		baseURL += `/u/${ userEmail }`;
```

Comment:
Should we be urlencoding this email? Or does that happen elsewhere?

Reply:
Is that necessary here for an email? Would it ever have special characters?

Reply:
In my testing it doesn't seem to be necessary when used here or with `authuser` param.

Reply:
I mean, validating emails is a can of worms we don't need to get too into 😆

But my understanding is you could have an email with non-ASCII text: https://en.wikipedia.org/wiki/Email_address#Internationalization

So an email like `шел+testлы@foo.com` would result in the URL `https://www.google.com/adsense/new/u/%D1%88%D0%B5%D0%BB+test%D0%BB%D1%8B@foo.com` instead of `https://www.google.com/adsense/new/u/шел+testлы@foo.com`.

Maybe not a big deal though? 🤷

Reply:
Let's leave as is for now, we can re-evaluate if it results in problems.